lib/file_store/adapters/memory.ex

```diff
@@ -130,6 +130,25 @@ defmodule FileStore.Adapters.Memory do
       end)
     end
 
+    def stream!(store, key, opts \\ []) do
+      Agent.get(store.name, fn state ->
+        case Map.fetch(state, key) do
+          :error ->
+            raise FileStore.StreamError, reason: "file does not exist", key: key
```

Comment:
This is a `!` function. Is this the right error to raise here? The error adapter won't be able to handle this with its `wrap` function.

Reply:
I was trying to emulate / stick close to the `File` implementation that only supports `stream!`. A non-bang method does not exist on that interface. But I don't see why we couldn't add it here. It would just mean for the `Disk` adapter, the non-bang `stream` could actually raise. Not the end of the world.

Reply:
Right, I don't think this API needs to match the File API. There are different concerns here. The disk adapter could rescue exceptions from File.stream, if need be.

I feel pretty comfortable letting you decide what API makes the most sense.

Reply:
> I guess it's kind of confusing if the return signature includes `{:error, reason}`, but an exception could still be raised.

Fair point.